src/platform/Darwin/UserDefaults.h

```diff
@@ -0,0 +1,26 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+namespace chip {
+namespace DeviceLayer {
+
+uint16_t getUserDefaultDnssdSRPTimeoutInMSecs();
```

Comment:
This needs to document that 0 means no value or something.

Or better yet return `std::optional<uint16_t>` and then the caller can use `value_or`.  That would incidentally allow overriding with 0, which might be desirable.